test/plugins/fingerprint.pulgin.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { VerifyFiltersDto } from '../../src/plugins/dto/verify.filters.dto';
+import { FingerprintPlugin } from '../../src/plugins/impl/fingerprint.plugin';
+import { MockBioAuthService } from '../mock/mock.bio.auth.service';
+import { ExternalId } from '../../src/db/entity/external.id';
+import { MockRepository } from '../mock/mock.repository';
+import { now, pepperHash } from '../support/functions';
+import { VerifyFingerprintImageDto } from '../../src/plugins/dto/verify.fingerprint.image.dto';
+import { ProtocolException } from 'protocol-common/protocol.exception';
+import { ProtocolErrorCode } from 'protocol-common/protocol.errorcode';
+
+export class MockFingerpintPlugin implements VerifyFiltersDto {
+
+    constructor(private readonly status: string, private readonly id: string, private readonly agentId: string) {}
+
+    externalIds: object;
+    async getIds(filters: VerifyFiltersDto): Promise<Map<string, string>> {
+        return new Map<string, string>(Object.entries(filters.externalIds));
+    }
+
+    async verify(params: any, filters: VerifyFiltersDto): Promise<{}> {
+        return Promise.resolve({
+            data: {
+                status: this.status,
+                did: this.id,
+            },
+        });
+    }
+
+    async verifyFingerprint(params: any, position: number, image: string, agentId: string): Promise<any> {
+        return Promise.resolve({
+            data: {
+                status: this.status,
+                agentId: this.agentId,
+            },
+        });
+    }
+}
+describe('Fingerprint Plugin is valid', () => {
+    //const fingerprintVerifySpec = new VerifyFiltersDto ;
+    it('Plugin is not empty', async () => {
+        const agentId = 'agentId123';
+        const bioAuthService = new MockBioAuthService('not_matched', agentId);
+
+        const id2 = 1000000 + parseInt(now().toString().substr(7,6), 10);
+        const id1 = 'N' + id2;
+        const id1Hash = pepperHash(id1);
+        const mockExternalId = new ExternalId();
+        mockExternalId.agent_id = agentId;
+        mockExternalId.external_id = id1Hash;
+        mockExternalId.external_id_type = 'id_1';
+        const mockExternalIdRepository: any = new MockRepository<ExternalId>([mockExternalId]);
+        
+        const fpPlugin = new FingerprintPlugin(bioAuthService, mockExternalIdRepository);
+
+        const params = {
+            position: 1,
+            image: 'some image'
+        } as VerifyFingerprintImageDto;
+        const filters = {
+            externalIds: {
+                key: 'value'
+            }
+        } as VerifyFiltersDto;
+        (await expect(fpPlugin.verify(params, filters))).rejects.toThrow(ProtocolException);
+
+    });
+    it('If fingerpirnt no match is thrown should throw a Protocol exception', async () => {
```

Comment:
Typo: "fingerpirnt" should be "fingerprint"

Also, it's helpful to use the exact name of things in the title. So instead of "fingerprint no match", use "FINGERPRINT_NO_MATCH". And instead of "Protocol exception", use "ProtocolException".